game/main.cc

```diff
@@ -66,43 +66,45 @@ static void signalSetup() {
 /// can be thrown as an exception to quit the game
 struct QuitNow {};
 
-static void checkEvents(Game& gm) {
+static void checkEvents(Game& gm, Time eventTime) {
 	if (g_quit) {
 		std::cerr << "Terminating, please wait... (or kill the process)" << std::endl;
 		throw QuitNow();
 	}
+	Window& window = gm.window();
 	SDL_Event event;
-	while(SDL_PollEvent(&event) == 1) {
+	while (SDL_PollEvent(&event) == 1) {
 		// Let the navigation system grab any and all SDL events
-		auto eventTime = Clock::now();
 		gm.controllers.pushEvent(event, eventTime);
-		switch(event.type) {
-		  case SDL_QUIT:
-			gm.finished();
-			break;
-		  case SDL_KEYDOWN: {
-			int keypressed  = event.key.keysym.scancode;
-			uint16_t modifier = event.key.keysym.mod;
-			if (((keypressed == SDL_SCANCODE_RETURN || keypressed == SDL_SCANCODE_KP_ENTER) && modifier & KMOD_ALT) || keypressed == SDL_SCANCODE_F11) {
+		auto type = event.type;
+		if (type == SDL_WINDOWEVENT) {
+			auto t = event.window.event;
+#ifdef __APPLE__
+			// Sent by MacOS green button (full screen mode)
```

Comment:
Please use the platform class that can even provide runtime checks to see whether we're running on mac.

Otherwise, I think it's ok but cannot test it for now.